### ISSUE TITLE
Ability to use version from registry as latestVersion

### ISSUE THREAD

**Issue by nordluf** [release-it/release-it#548]
Hello,

I'm switching from version 7 and try to get the same behavior I have there: conventionally (angular) bumping version based on already done releases, not one in the package.json (I always have 0.0.0 in package.json). 

My previous .release-it.json looks like:
`{
  "increment": "conventional:angular",
  "requireCleanWorkingDir": false,
  "src": {
    "commit": false,
    "tagName": "v%s"
  }
}
`

So my workflow is:
- I have version 1.10.3 published (a version in the package.json is 0.0.0 and no CHANGELOG or VERSIONS)
- I do a commit 'feat: adding new functionality'
- version bumpings to 1.11.0 and then releases

Is it possible with latest release-it?

**Comment by webpro** [comment]
Should not be a problem:

* See https://github.com/release-it/release-it/blob/master/docs/changelog.md#conventional-changelog
* Use `--no.npm` to skip package.json & npm actions.

**Comment by nordluf** [comment]
I still get 0.1.0 as a version for publishing, even with --no.npm 
Also, --no.npm sounds odd for me - I do want to use npm to publish a release. 
The only problem is - I want or calculate version based on commits from the scratch, or by bumping latest registry version.

**Comment by webpro** [comment]
Sorry, typo. It should be `--no-npm`.

I don't think you can publish to npm if the version in package.json stays 0.0.0?

**Comment by nordluf** [comment]
You are right, I can't publish with version 0.0.0 that's why I want to bump version before. Based on my commit history calculated version should be 1.11.0 and I want to publish such version.

**Comment by nordluf** [comment]
By using the correct option `--no-npm` I get correct version 1.11.0, but no npm release then

**Comment by webpro** [comment]
In this case you can start out with `release-it 1.11.0`. After this, version in package.json is correct, and bumps are based on commit history with the plugin.

**Comment by nordluf** [comment]
I use it in CI environment, that means - I can't set the correct version manually every time. 

There is a method of NPM plugin in the sources: getLatestRegistryVersion. Now, it uses only to show a warning that registry version doesn't match package.json version. Is it possible to have an option to use it as getLatestVersion

With this patch, it works as I expect it to work
https://github.com/release-it/release-it/blob/master/lib/plugin/npm/npm.js


`58,60d57
<         if (semver.lt(latestVersion, latestVersionInRegistry)) {
<           this.setContext({latestVersionInRegistry});
<         }
70c67
<     return this.getContext('latestVersionInRegistry') || this.getContext('latestVersion');
---
>     return this.getContext('latestVersion');
`

**Comment by webpro** [comment]
> I use it in CI environment, that means - I can't set the correct version manually every time.

You would need to do it only once. After this, the `npm` and `conventional-changelog` plugins will do their jobs.

> With this patch, it works as I expect it to work

This tool is used by more people. Your patch may lead to unexpected behavior. It's kinda strange that a local `package.json` has `0.0.0`, while the registry has the correct version. Should release-it fix this for you?

**Comment by nordluf** [comment]
> This tool is used by more people. 
That's why I proposed to use that by providing a setting in config file. The code itself was to show the behavior. 

Well, anyway I have found a workaround. If my proposal doesn't sound reasonable - close the issue

**Comment by webpro** [comment]
Could you please share your workaround? If I understand the problem better, maybe we can improve the documentation, implement a new feature, etc.

**Comment by nordluf** [comment]
Well, actually it is pretty simple: 


```
{
  "git": {
    "requireCleanWorkingDir": false,
    "commit": false,
    "tag": true,
    "tagName": "v${version}"
  },
  "plugins": {
    "@release-it/conventional-changelog": {
      "preset": "angular"
    },
    "@release-it/bumper": {
      "in": {
        "file": "VERSION",
        "type": "text/plain"
      }
    }
  },
  "hooks": {
    "before:init": "git tag --list 'v*' | sort -V | tail -1 > VERSION"
  }
}
```

**Comment by nordluf** [comment]
I'd like to see a builtin functionality, instead of that hack, but I'm not the decision-maker here.

**Comment by webpro** [comment]
I've added `--npm.ignoreVersion` to take the version from the latest Git tag in [v12.6.0](https://github.com/release-it/release-it/releases/tag/12.6.0).
